### PR TITLE
Single End Support

### PR DESCRIPTION
**Pull request by Hammarn**
- Support for running with Single End
- Dockerfile cleanup
- `fusion_genes_compare.py` now uses a generic file grouping instead of NGI-filenames
	- Also a small bugfix which occured when only one fusion was found.

### CODE REVIEW

File: bin/fusion_genes_compare.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import argparse
 import sys
 
+import pdb
```
Comment (ewels) [inline bin/fusion_genes_compare.py:7]: oops

File: bin/fusion_genes_compare.py [hunk 2]
```diff
@@ -25,6 +26,7 @@ def read_files_store_data(input_files,output_file):
 
         elif input_file.endswith("summary_candidate_fusions.txt"):
             #We have a Fusion catcher file
+            pdb.set_trace()
```
Comment (ewels) [inline bin/fusion_genes_compare.py:29]: oops

File: bin/fusion_genes_compare.py [hunk 3]
```diff
@@ -84,16 +95,19 @@ def make_report(fusion_dict, output_file):
     content+="## Number of Fusion genes detected with FusionCatcher: {} \n".format(len_fc)
     content +="##FUSIONCATCHER\tSTAR-FUSION\tBOTH\n"
     ##cleanup
+    
+    
     gene_in_both=[item.rstrip() for item in gene_in_both]
     gene_star_only=[item.rstrip() for item in gene_star_only]
     gene_fc_only=[item.rstrip() for item in gene_fc_only]
     
+    pdb.set_trace()
```
Comment (ewels) [inline bin/fusion_genes_compare.py:104]: oops

File: main.nf [hunk 4]
```diff
@@ -24,19 +24,35 @@ version = '0.1'
...
     .into { read_files_star_fusion; fusion_inspector_reads; fusioncatcher_reads}
 
 
+// Validate inputs
+if( params.star_fusion_reference && params.star_fusion ){
+    star_fusion_reference = Channel
+        .fromPath(params.star_fusion_reference)
+        .ifEmpty { exit 1, "STAR-fusion reference not found: ${params.star_fusion_reference}" }
+}
+(star_fusion_reference,star_fusion_reference_fusioninspector) = star_fusion_reference.into(2)
```
Comment (ewels) [inline main.nf:48]: Probably nicer to do this instead:

```groovy
Channel.fromPath(params.star_fusion_reference)
    .ifEmpty { exit 1, "STAR-fusion reference not found: ${params.star_fusion_reference}" }
    .into { star_fusion_reference, star_fusion_reference_fusioninspector }
```

Comment (ewels) [reply main.nf:48]: Also, I guess that having this line outside of the `if` statement will crash the pipeline if the `if` evaluates to false?

File: main.nf [hunk 5]
```diff
@@ -109,34 +91,50 @@ process fusioncatcher {
...
     file '*.{txt,log,zip}' into fusioncatcher
-
     when: params.fusioncatcher
 
     script:
+    if (params.singleEnd) {
+    
     """
+    mkdir ${reads}_data
+    mv ${reads} ${reads}_data/
```
Comment (ewels) [inline main.nf:105]: Is the directory name important? May be safer to just call it `data`..

**Review by ewels — COMMENTED** [review]
Sorry for missing this review! Spotted a few things, can take them up in another PR.

File: main.nf (hunk 5 above)
Comment (Hammarn) [reply main.nf:105]: Yes, I think so. If I recall correctly a lot of the output it always called the same and then it's nice to have the sample name in the directory to know what you are looking at. Especially while debugging. I think..

File: main.nf (hunk 4 above)
Comment (Hammarn) [reply main.nf:48]: Yeah, I think you are right.
But I assume your code will casue iddues as well. Should be no need to have a star fusion reference if we are not running that process!?

Comment (ewels) [reply main.nf:48]: Sure - I meant that my code would be inside the `if` sorry.

File: main.nf (hunk 5 above)
Comment (ewels) [reply main.nf:105]: Fair enough 👍

**Pull request merged**